nodejs/aws-infra/experimental/loadBalancer.ts

```diff
@@ -0,0 +1,254 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as docker from "@pulumi/docker";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "../utils";
+
+import * as mod from ".";
+
+export type LoadBalancers = aws.ecs.ServiceArgs["loadBalancers"];
+
+export interface ILoadBalancerProvider {
+    portMappings(containerName: string, name: string, parent: pulumi.Resource): pulumi.Input<aws.ecs.PortMapping[]>;
+    loadBalancers(containerName: string, name: string, parent: pulumi.Resource): LoadBalancers;
```

Comment:
as above, Task/Service doesn't have to know/care about how these are implemented.  they simply return the data appropriate for them to add to their specific configs.